fields/types/file/FileField.js

```diff
@@ -56,15 +58,15 @@ module.exports = Field.create({
 	getFilename () {
 		return this.state.userSelectedFile
 			? this.state.userSelectedFile.name
-			: this.props.value.filename;
+			: this.props.value && this.props.value.filename;
```

Comment:
I think we're solving this the wrong way, props.value should _always_ be an object, and we should fix that somewhere higher than this to reduce code branches.

The probably correct but more-work way would be to dispatch new item values / nested item values from something central that can get the default value from each field type (which would take into account field defaults)

Quicker way is putting value in state and updating it with defaults correctly applied in `componentWillUpdate`, maybe?